pkg/controller/podautoscaler/horizontal.go

```diff
@@ -211,10 +213,15 @@ func (a *HorizontalController) processNextWorkItem() bool {
 	if err != nil {
 		utilruntime.HandleError(err)
 	}
-	// Add request processing HPA after resync interval just in case last resync didn't insert
-	// request into the queue. Request is not inserted into queue by resync if previous one wasn't processed yet.
-	// This happens quite often because requests from previous resync are removed from the queue at the same moment
-	// as next resync inserts new requests.
+	// Add request processing HPA to be executed after resyncPeriod. Requests are always added to
+	// queue to be executed after resyncPeriod. If there's already request for the HPA in the queue
+	// then a new request is always dropped. Requests spend resyncPeriod in queue
+	// so HPAs are processed every resyncPeriod.
+	// Requests is added here just in case last resync didn't insert request into the queue. This
+	// happens quite often because there is race condition between adding request after resyncPeriod
+	// and removing them from queue. Request can be added by resync before previous request is
```

Comment:
"Request can be added by resync before previous request is
	// processed. In this case request to be executed after resyncPeriod is dropped and HPA is processed
	// after 2 x resyncPeriod."

Is that what happens right now. Or what would happen without the logic below?

Reply:
Request created by rsync are dropped but we add request here to avoid processing hpa after 2xresyncPeriod.